ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -286,9 +286,9 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, instance
 			}
 
 			// We can start the Redis cleanup process now because the head Pod has been terminated.
-			filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: instance.Name, utils.RayNodeTypeLabelKey: string(rayv1.RedisCleanupNode)}
+			filterLabels := common.RayClusterRedisPodsAssociationOptions(instance).ToListOptions()
```

Comment:
1. It is used to associate a K8s Job instead of Pod.
2. It is not a Redis Pod.

Please rename to `RayClusterRedisCleanupJobAssociationOptions`